lib/bitcoin/interfaces/IBitcoinWallet.ts

```diff
@@ -0,0 +1,36 @@
+import { Address, Script, Transaction } from 'bitcore-lib';
+
+/**
+ * Represents a bitcoin wallet.
+ */
+export default interface IBitcoinWallet {
+
+  /**
+   * Gets the public key associated with this wallet.
+   *
+   * @returns The public key associated with this wallet as Buffer.
+   */
+  getPublicKey (): Buffer;
+
+  /**
+   * Gets the address objecr associated with this wallet.
+   */
+  getAddress (): Address;
+
+  /**
+   * Signs the specified transaction using the keys associated with this wallet.
+   *
+   * @param sidetreeTransaction The sidetree transaction to sign.
+   * @returns The signed transaction.
+   */
+  signTransaction (sidetreeTransaction: Transaction): Promise<Transaction>;

```

Comment:
Nit: Maybe `signSidetreeTransaction` to disambiguate from bitcoin transactions.

Reply:
Yes it should be transaction. This is a leftover for when I had a different function name.